data/modules/CargoRun/CargoRun.lua

```diff
@@ -0,0 +1,1148 @@
+			domicile	= ad.domicile,
+			client		= ad.client,
+			location	= ad.location,
+			localdelivery	= ad.localdelivery,
+			wholesaler	= ad.wholesaler,
+			pickup		= ad.pickup,
+			cargo_picked_up	= cargo_picked_up,
+			introtext	= ad.introtext,
+			risk		= ad.risk,
+			reward		= ad.reward,
+			due		= ad.due,
+			cargo		= ad.cargo,
+			branch		= ad.branch,
+			cargotype	= ad.cargotype,
+		}
+		table.insert(missions,Mission.New(mission))
+		if ad.pickup then
+			form:SetMessage(l["ACCEPTED_PICKUP_" .. Engine.rand:Integer(1, getNumberOf("ACCEPTED_PICKUP"))])
+		else
+			form:SetMessage(l["ACCEPTED_" .. Engine.rand:Integer(1, getNumberOf("ACCEPTED"))])
+		end
+		return
+
+	elseif option == 4 then
+		local num = math.floor(ad.urgency * (getNumberOf("URGENCY") - 1)) + 1
+		local urgency = l["URGENCY__" .. ad.branch .. "__" .. ad.cargotype] or l["URGENCY__" .. ad.branch] or l["URGENCY_" .. num]
+		form:SetMessage(urgency .. Format.Date(ad.due))
+
+	elseif option == 5 then
+		if ad.localdelivery then -- very low risk -> no specific text to give no confusing answer
+			form:SetMessage(l.RISK_1)
+		else
+			local num = math.floor(ad.risk * (getNumberOf("RISK") - 1)) + 1
+			local risk = l["RISK__" .. ad.branch .. "__" .. ad.cargotype] or l["RISK__" .. ad.branch] or l["RISK_" .. num]
+			if ad.wholesaler and ad.introtext ~= HABER then
+				risk = risk .. " " .. l.RISK_WHOLESALER
+			end
+			form:SetMessage(risk)
+		end
+	end
+
+	form:AddOption(l.WHY_SO_MUCH_MONEY, 1)
+	form:AddOption(l.HOW_MUCH_MASS, 2)
+	form:AddOption(l.HOW_SOON_MUST_IT_BE_DELIVERED, 4)
+	form:AddOption(l.WILL_I_BE_IN_ANY_DANGER, 5)
+	form:AddOption(l.COULD_YOU_REPEAT_THE_ORIGINAL_REQUEST, 0)
+	form:AddOption(l.OK_AGREED, 3)
+end
+
+local onDelete = function (ref)
+	ads[ref] = nil
+end
+
+local isEnabled = function (ref)
+	return isQualifiedFor(Character.persistent.player.reputation, ads[ref])
+end
+
+local findNearbyStations = function (station, minDist)
+	local nearbystations = {}
+	for _,s in ipairs(Game.system:GetStationPaths()) do
+		if s ~= station.path then
+			local dist = station:DistanceTo(Space.GetBody(s.bodyIndex))
+			if dist >= minDist then
+				table.insert(nearbystations, { s, dist })
+			end
+		end
+	end
+	return nearbystations
+end
+
+local nearbysystems
+
+local makeAdvert = function (station, freeCargoSpace)
+	local reward, due, location, nearbysystem, dist, nearbystations, cargo, adtext, introtext
+	local risk, wholesaler, pickup
+	local client = Character.New()
+	local urgency = Engine.rand:Number(0, 1)
+	local localdelivery = Engine.rand:Number(0, 1) > 0.5 and true or false
+	local branch = Engine.rand:Integer(1, #custom_cargo)
+	local cargotype = Engine.rand:Integer(1, #custom_cargo[branch])
+
+	if localdelivery then
+		nearbysystem = Game.system
+		nearbystations = findNearbyStations(station, 1000)
+		if #nearbystations == 0 then return nil end
+		if freeCargoSpace ~= nil then
+			cargo = freeCargoSpace == 1 and 1 or Engine.rand:Integer(1, freeCargoSpace)
```

Comment:
You don't need to tree the specail case of `freeCargoSpace==1`, since `Engine.rand:Integer(1, 1)` will return 1.